### ISSUE TITLE
Trying out inference

### ISSUE THREAD

**Issue by salman-moh** [MolecularAI/Chemformer#10]
Hi @EBjerrum , I followed the steps in creating new venv and also downloaded the data/models from the link hosted by AZ.
I'm assuming I could try running [predict.py](https://github.com/MolecularAI/Chemformer/blob/main/molbart/predict.py) and obtain the results connecting model path and the dataset path in the args.
However, I have a problem understanding what the args mean.
![image](https://user-images.githubusercontent.com/31631107/194123628-9b8add72-51cd-4883-8f98-8c36754380cb.png)
`--reactants_path` : Is this the pickle file dataset USPTO_50 for example? but the code I think expects a text file
`--model_path` : the path here I have is from the downloaded pretrained model 'combined' version
`--products_path` : the path where dataframe is saved on
`--vocab_path` : the path of bart_vocab
The rest of them, just using default values.
Would like to get this working to put up a working dashboard on HF Space :) Thanks!

**Comment by anniewesterlund** [comment]
Hi Salman!

* The example_scripts/predict.sh shows some information about the input types:

python -m molbart.predict \
  --reactants_path uspto_50_test.txt \
  --products_path uspto_50_p0_5_out.pickle \
  --model_path saved_models/uspto_50/span_aug/100_epochs/last.ckpt \
  --batch_size 64 \
  --num_beams 10


* Some explanations:
--reactants_path : Is the text file dataset with e.g. USPTO_50. Each line is an input SMILES. The content of this text file may look something like: 
```
C1=COCCC1.COC(=O)CCC(=O)c1ccc(O)cc1O
COC(=O)c1cccc(C(=O)O)c1.Nc1cccnc1N
CC(C)(C)OC(=O)NC1CCC(C(=O)O)CC1.CNOC
Nc1ccc(O)cc1.O=[N+]([O-])c1ccc(Cl)nc1Cl
```

--model_path : Path to the pre-trained model
--products_path : Path to the output dataframe (pickle).
--vocab_path: Path to the bart_vocab, e.g. "bart_vocab.txt"

Hope this helps!
/Annie

**Comment by salman-moh** [comment]
Hi Annie, thanks for your reply.

However, I dont find the text file they mention in example scripts for `--reactants_path` uspto_50_test.txt
Have you tested out this model by yourself? Would like to know where you get the text files from!

**Comment by SGenheden** [comment]
There are many USPTO datasets you can find on the Internet.

- Here is the one called USPTO-MIT: https://github.com/wengong-jin/nips17-rexgen
- There is also the number of datasets used with the original MolecularTransformer from this paper: https://doi.org/10.1021/acscentsci.9b00576, data can be downloaded from here: https://ibm.ent.box.com/v/MolecularTransformerData

**Comment by salman-moh** [comment]
Thanks for your reply Samuel!

I am currently using the validation set (on USPTO-15K) to do predictions (as it was the smallest file). With the following contents(first 2 lines):

```
[CH3:18][CH2:19][O:20][C:21](=[O:22])[Cl:23].[CH3:1][O:2][c:3]1[cH:4][c:5]2[n:6][c:7]([NH2:8])[c:9]([O:10][CH3:11])[n:12][c:13]2[cH:14][c:15]1[O:16][CH3:17].[cH:24]1[cH:25][cH:26][n:27][cH:28][cH:29]1>>CCOC(=O)Nc1nc2cc(OC)c(OC)cc2nc1OC 8;;21-23-1.0;21-8-1.0
[CH3:1][C:2]([CH3:3])([CH3:4])[O:5][C:6](=[O:7])[c:8]1[cH:9][cH:10][c:11](-[n:12]2[c:13]([CH2:14][CH2:15][C:16](=[O:17])[OH:18])[cH:19][cH:20][c:21]2-[c:22]2[cH:23][cH:24][cH:25][cH:26][cH:27]2)[cH:28][cH:29]1.[CH3:43][C:44](=[O:45])[O-:46].[NH4+:42].[O:30]=[C:31]([n:32]1[cH:33][cH:34][n:35][cH:36]1)[n:37]1[cH:38][cH:39][n:40][cH:41]1>>CC(C)(C)OC(=O)c1ccc(-n2c(CCC(N)=O)ccc2-c2ccccc2)cc1 ;32;16-18-1.0,31-32-1.0,32-33-1.5,32-36-1.5;16-32-1.0
```
The evaluation is running atm, once its done successfully I will close this issue if that is ok with you :)

**Comment by salman-moh** [comment]
This is the output on USPTO-15K dataset; a few questions on the output (I'm new to the pharma domain)
1. looks like this is a Retrosynthesis prediction task?
2. but whats with the question marks in the predictions?
3. if it is indeed retrosynthesis prediction, why are there exactly 10 prediction simpler molecules for all? I would think some molecules from the USPTO-15K dataset can be broken down into maybe less than 10 simpler molecules as opposed to broken down into always 10 simpler molecules. 

The columns of the output file: 
> | original_smiles | prediction_0 | log_likelihood_0 | prediction_1 | log_likelihood_1 | prediction_2 | log_likelihood_2 | prediction_3 | log_likelihood_3 | prediction_4 | log_likelihood_4 | prediction_5 | log_likelihood_5 | prediction_6 | log_likelihood_6 | prediction_7 | log_likelihood_7 | prediction_8 | log_likelihood_8 | prediction_9 | log_likelihood_9

![image](https://user-images.githubusercontent.com/31631107/194644356-3add1b15-7677-4cf2-913c-3d3cf8233787.png)

**Comment by salman-moh** [comment]
Worked with the above suggestions!

**Comment by llmgpt** [comment]
Why are my results garbled?
![2024-09-25 13-29-59 的屏幕截图](https://github.com/user-attachments/assets/001a2ba1-1d76-4f5e-8f2c-55ad4ec82260)

**Comment by llmgpt** [comment]
> Hi Salman!
> 
>     * The example_scripts/predict.sh shows some information about the input types:
> 
> 
> python -m molbart.predict --reactants_path uspto_50_test.txt --products_path uspto_50_p0_5_out.pickle --model_path saved_models/uspto_50/span_aug/100_epochs/last.ckpt --batch_size 64 --num_beams 10
> 
>     * Some explanations:
>       --reactants_path : Is the text file dataset with e.g. USPTO_50. Each line is an input SMILES. The content of this text file may look something like:
> 
> 
> ```
> C1=COCCC1.COC(=O)CCC(=O)c1ccc(O)cc1O
> COC(=O)c1cccc(C(=O)O)c1.Nc1cccnc1N
> CC(C)(C)OC(=O)NC1CCC(C(=O)O)CC1.CNOC
> Nc1ccc(O)cc1.O=[N+]([O-])c1ccc(Cl)nc1Cl
> ```
> 
> --model_path : Path to the pre-trained model --products_path : Path to the output dataframe (pickle). --vocab_path: Path to the bart_vocab, e.g. "bart_vocab.txt"
> 
> Hope this helps! /Annie

Please tell me what is the relationship between these four chemical molecules?

**Comment by EBjerrum** [comment]
It's four different inputs. Reactants seperated with a .

**Comment by llmgpt** [comment]
Thank you for your reply.
